### ISSUE TITLE
Flaky test: ShutdownTests.ServerShutsDownWhenMainExits

### ISSUE THREAD

**Issue by jkotalik** [aspnet/IISIntegration#995]
```
System.Net.Http.HttpRequestException : No connection could be made because the target machine actively refused it.
---- System.Net.Sockets.SocketException : No connection could be made because the target machine actively refused it.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IntegrationTesting.LoggingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/test/IISIntegration.FunctionalTests/Utilities/LoggingHandler.cs:line 25
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/test/IISIntegration.FunctionalTests/Utilities/RetryHandler.cs:line 36
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests.ServerShutsDownWhenMainExits() in /_/test/IISIntegration.FunctionalTests/Inprocess/ShutdownTests.cs:line 27
--- End of stack trace from previous location where exception was thrown ---
----- Inner Stack Trace -----
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
------- Stdout: -------
| [0.000s] TestLifetime Information: Starting test ServerShutsDownWhenMainExits at 2018-06-28T21:38:40
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Deploying [Variation] :: ServerType=IISExpress, Runtime=CoreClr, Arch=x64, BaseUrlHint=, Publish=True
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Removing environment variable ASPNETCORE_ENVIRONMENT
| [0.004s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Executing command dotnet publish  --output "C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e" --framework netcoreapp2.2 --configuration Release --no-restore -p:VerifyMatchingImplicitPackageVersion=false 
| [0.211s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout: Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
| [0.211s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout: Copyright (C) Microsoft Corporation. All rights reserved.
| [0.992s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout:   TestTasks -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\test\TestTasks\bin\Release\netcoreapp2.2\TestTasks.dll
| [1.069s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout:   Microsoft.AspNetCore.Server.IIS -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\src\Microsoft.AspNetCore.Server.IIS\bin\Release\netstandard2.0\Microsoft.AspNetCore.Server.IIS.dll
| [1.446s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout:   InProcessWebSite -> C:\BuildAgent\work\33bdfc1cae7b2a38\modules\IISIntegration\test\WebSites\InProcessWebSite\bin\Release\netcoreapp2.2\InProcessWebSite.dll
| [2.112s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet-publish stdout:   InProcessWebSite -> C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\
| [2.155s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: dotnet publish finished with exit code : 0
| [2.156s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Attempting to start IIS Express on port: 55552
| [2.156s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Writing '[ANCMPath]' 'C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\x64\aspnetcore.dll' to config
| [2.156s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Writing '[ANCMV2Path]' 'C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\x64\aspnetcorev2.dll' to config
| [2.157s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Writing [PORT] '55552' to config
| [2.157s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Writing [ApplicationPhysicalPath] 'C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e' to config
| [2.166s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Saving Config to C:\BuildAgent\temp\buildTmp\tmpA58D.tmp
| [2.168s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Executing command : C:\Program Files\IIS Express\iisexpress.exe /site:HttpTestSite /config:C:\BuildAgent\temp\buildTmp\tmpA58D.tmp /trace:error
| [2.168s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Removing environment variable ASPNETCORE_ENVIRONMENT
| [2.168s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: SET ASPNETCORE_DETAILEDERRORS=true
| [2.168s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: SET ASPNETCORE_MODULE_DEBUG=4
| [2.170s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress Process 11712 started
| [2.217s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Starting IIS Express ...
| [2.230s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Initializing the W3 Server Started CTC = 1254204843 
| [2.235s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: W3 Server initializing WinSock.  CTC = 1254204859 
| [2.235s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: W3 Server WinSock initialized.  CTC = 1254204859 
| [2.235s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: W3 Server ThreadPool initialized (ipm has signalled).  CTC = 1254204859 
| [2.258s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Start listenerChannel http:0
| [2.259s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Successfully registered URL "http://localhost:55552/" for site "HttpTestSite" application "/"
| [2.259s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Registration completed for site "HttpTestSite"
| [2.260s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: AppPool 'IISExpressAppPool' initialized
| [2.260s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: InitComplete event signalled
| [2.260s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: IIS Express is running.
| [2.260s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Started iisexpress successfully. Process Id : 11712, Port: 55552
| [2.261s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Application ready at URL: http://localhost:55552/
| [2.262s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Method: GET, RequestUri: 'http://localhost:55552/Shutdown', Version: 2.0, Content: <null>, Headers:
|                                                                                          {
|                                                                                          }
| [2.272s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: IncrementMessages called
| [2.274s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Request started: "GET" http://localhost:55552/Shutdown
| [2.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Resolving hostfxr parameters for application: dotnet arguments: .\InProcessWebSite.dll path: C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\
| [2.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Process path C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\dotnet is dotnet, treating application as portable
| [2.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Resolving absolute path do dotnet.exe from C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\dotnet
| [2.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Invoking where.exe to find dotnet.exe
| [2.415s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] where.exe invocation returned: C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: C:\Program Files\dotnet\dotnet.exe
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Current process bitness type detected as isX64=1
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Processing entry C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Binary type 6
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Found dotnet.exe wia where.exe invocation at C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Resolving absolute path do hostfxr.dll from C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] hostfxr.dll located at C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\host\fxr\2.2.0-preview1-26618-02\hostfxr.dll
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Resolving hostfxr_main arguments application: C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe arguments: .\InProcessWebSite.dll path: C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\
| [2.416s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Argument[1] = C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\InProcessWebSite.dll
| [2.443s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcore.dll] Loading request handler: C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\x64\aspnetcorev2_inprocess.dll
| [2.447s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] REQUESTHANDLER_CONFIG::GetConfig, set config to ModuleContext
| [2.448s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Resolving hostfxr parameters for application: dotnet arguments: .\InProcessWebSite.dll path: C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\
| [2.448s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Process path C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\dotnet is dotnet, treating application as portable
| [2.448s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Resolving absolute path do dotnet.exe from C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\dotnet
| [2.448s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Invoking where.exe to find dotnet.exe
| [2.590s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] where.exe invocation returned: C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.590s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: C:\Program Files\dotnet\dotnet.exe
| [2.590s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Current process bitness type detected as isX64=1
| [2.590s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Processing entry C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Binary type 6
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Found dotnet.exe wia where.exe invocation at C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Resolving absolute path do hostfxr.dll from C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] hostfxr.dll located at C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\host\fxr\2.2.0-preview1-26618-02\hostfxr.dll
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Resolving hostfxr_main arguments application: C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe arguments: .\InProcessWebSite.dll path: C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\
| [2.591s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: [aspnetcorev2_inprocess.dll] Argument[1] = C:\BuildAgent\temp\buildTmp\fbf2abb4-a9f6-4da8-840d-f318bd152b4e\InProcessWebSite.dll
| [2.995s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Request ended: http://localhost:55552/Shutdown with HTTP status 200.0
| [2.995s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnection+HttpConnectionResponseContent, Headers:
|                                                                                          {
|                                                                                            Transfer-Encoding: chunked
|                                                                                            Server: Microsoft-IIS/10.0
|                                                                                            X-SourceFiles: =?UTF-8?B?QzpcQnVpbGRBZ2VudFx0ZW1wXGJ1aWxkVG1wXGZiZjJhYmI0LWE5ZjYtNGRhOC04NDBkLWYzMThiZDE1MmI0ZVxTaHV0ZG93bg==?=
|                                                                                            X-Powered-By: ASP.NET
|                                                                                            Date: Thu, 28 Jun 2018 21:38:43 GMT
|                                                                                          }
| [3.024s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress Process 11712 shut down
| [3.110s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: iisexpress stdout: Request ended: http://localhost:55552/Shutdown with HTTP status 200.0
| [3.113s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Warning: Error sending request
| [3.114s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Retrying 1th time after 1 sec.
| [3.114s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: Host process shutting down.
| [4.114s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Method: GET, RequestUri: 'http://localhost:55552/Shutdown', Version: 2.0, Content: <null>, Headers:
|                                                                                          {
|                                                                                          }
| [6.330s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Warning: Error sending request
| [6.330s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Retrying 2th time after 1 sec.
| [7.316s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Method: GET, RequestUri: 'http://localhost:55552/Shutdown', Version: 2.0, Content: <null>, Headers:
|                                                                                          {
|                                                                                          }
| [9.301s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Warning: Error sending request
| [9.301s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Retrying 3th time after 1 sec.
| [10.303s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Method: GET, RequestUri: 'http://localhost:55552/Shutdown', Version: 2.0, Content: <null>, Headers:
|                                                                                           {
|                                                                                           }
| [12.280s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Warning: Error sending request
| [12.280s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Retrying 4th time after 1 sec.
| [13.281s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Debug: Method: GET, RequestUri: 'http://localhost:55552/Shutdown', Version: 2.0, Content: <null>, Headers:
|                                                                                           {
|                                                                                           }
| [15.274s] Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests.ShutdownTests Warning: Error sending request
| [15.275s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Warning: Host process already exited or never started successfully.
| [15.276s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Debug: Deleting applicationHost.config file from C:\BuildAgent\temp\buildTmp\tmpA58D.tmp
| [15.384s] Microsoft.AspNetCore.Server.IntegrationTesting.IISExpressDeployer Information: [Time]: Total time taken for this test variation '15.4794228' seconds
```

**Comment by pakrym** [comment]
Looks like server shuts down before sending the response.

**Comment by ryanbrandenburg** [comment]
ServerShutsDownWhenMainExits [failed again](http://aspnetci/viewLog.html?buildId=514782&buildTypeId=XPlat_Windows_Win2008r2_Universe).

**Comment by ryanbrandenburg** [comment]
ServerShutsDownWhenMainExits [failed again](http://aspnetci/viewLog.html?buildId=517407&buildTypeId=XPlat_Windows_Win2008r2_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
ServerShutsDownWhenMainExits [failed again](http://aspnetci/viewLog.html?buildId=517541&buildTypeId=XPlat_Windows_Win7_Universe).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by pakrym** [comment]
Should be fixed by #1120
